cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceTaskContext.java

```diff
@@ -192,8 +208,38 @@ public LoggingContext getLoggingContext() {
     return loggingContext;
   }
 
+  //---- following are methods to manage transaction lifecycle for the datasets. This needs to
+  //---- be refactored after [TEPHRA-99] and [CDAP-3893] are resolved.
+
+  private void initializeTransactionAwares() {
```

Comment:
So this method is just for starting TX over static datasets in the dataset cache, right?